tls/s2n_server_extensions.c

```diff
@@ -191,3 +203,14 @@ int s2n_recv_server_sct_list(struct s2n_connection *conn, struct s2n_stuffer *ex
 
     return 0;
 }
+
+int s2n_recv_server_max_frag_len(struct s2n_connection *conn, struct s2n_stuffer *extension)
+{
+    uint16_t max_fragment_length;
+    GUARD(s2n_stuffer_read_uint16(extension, &max_fragment_length));
+    if (max_fragment_length != mfl_code_to_length[conn->config->mfl_code]) {
+        S2N_ERROR(S2N_ERR_MAX_FRAG_LEN_MISMATCH);
```

Comment:
Colm, your mentioned scenario would apply in the case of the server receiving a fragment larger than what is negotiated from a client. (`client bug`)

However, the case protected by `S2N_ERROR(S2N_ERR_MAX_FRAG_LEN_MISMATCH)` is a **server bug** during `Server Hello Recv` step of TLS handshake where the server is not returning the correct max fragment length initially requested by the client. 

Server sending fragments larger than what client requested can break things on the users.